Move hal_nxp glue layer to zephyr repo (with leveraging MCUX SDK NEXT kconfig tree)

ZhaoxiangJin opened the pull request:
Version with leveraging MCUX SDK NEXT kconfig tree.

ZhaoxiangJin committed:
modules: hal_nxp: Move hal_nxp glue layer to zephyr repo

Move hal_nxp glue layer to zephyr repo.

Signed-off-by: Zhaoxiang Jin <Zhaoxiang.Jin_1@nxp.com>

ZhaoxiangJin committed:
west.yml: Update manifest to track hal_nxp side PR

Update manifest to track hal_nxp side PR

Signed-off-by: Zhaoxiang Jin <Zhaoxiang.Jin_1@nxp.com>

ZhaoxiangJin commented:
Hello @hakehuang, please help test, thanks.

carlescufi commented:
+1 from me, glue layers belong in the Zephyr tree, not the module repo.

hakehuang commented:
> Hello @hakehuang, please help test, thanks.

I kick a full build first, and let's see

ZhaoxiangJin committed:
soc: nxp/imx/imx8m: Add cortex-m core suffix items

Add cortex-m core suffix item for imx devices

Signed-off-by: Zhaoxiang Jin <Zhaoxiang.Jin_1@nxp.com>

ZhaoxiangJin committed:
boards: nxp: Set CONFIG_SECOND_CORE_MCUX to y in
second core kconfig file

CONFIG_SECOND_CORE_MCUX is not set to y when building
the RT1160 and RT1170 second core synchronization
examples. It is better to set this kconfig item to y at
the board layer.

Signed-off-by: Zhaoxiang Jin <Zhaoxiang.Jin_1@nxp.com>

ZhaoxiangJin committed:
soc: nxp: Remove unused struct members

The cntMode in gpc_tran_step_config_t is not used, removed it.

Signed-off-by: Zhaoxiang Jin <Zhaoxiang.Jin_1@nxp.com>

ZhaoxiangJin committed:
soc: nxp: Update rt7xx flash_init function

The XSPI features was updated in MCUXSDK, need to update
zephyr side code to avoiding build error.

Signed-off-by: Zhaoxiang Jin <Zhaoxiang.Jin_1@nxp.com>

ZhaoxiangJin commented:
> > Hello @hakehuang, please help test, thanks.
> 
> I kick a full build first, and let's see

Hello @hakehuang, I have fixed all twister issues reported by CI, could you please help test again? Thanks.

hakehuang commented:
> Hello @hakehuang, I have fixed all twister issues reported by CI, could you please help test again? Thanks.

@ZhaoxiangJin what is the relation with this pr? https://github.com/zephyrproject-rtos/zephyr/pull/84423 they get the same name

ZhaoxiangJin committed:
MAINTAINERS: Remove some files from MAINTAINERS

Kconfig.imx, Kconfig.mcux, Kconfig.nxp_s32 have moved
inside hal_nxp, need to remove them from MAINTAINERS.yml

Signed-off-by: Zhaoxiang Jin <Zhaoxiang.Jin_1@nxp.com>

ZhaoxiangJin commented:
> > Hello @hakehuang, I have fixed all twister issues reported by CI, could you please help test again? Thanks.
> 
> @ZhaoxiangJin what is the relation with this pr? #84423 they get the same name, also please fix below:
> 
> ```
> get_maintainer.MaintainersError: /home/runner/work/zephyr/zephyr/MAINTAINERS.yml: glob pattern 'modules/Kconfig.imx' in 'files' in area 'West project: hal_nxp' does not match any files
> 
> Exception occurred:
>   File "/home/runner/work/zephyr/zephyr/scripts/get_maintainer.py", line 593, in _err
>     raise MaintainersError(msg)
> get_maintainer.MaintainersError: /home/runner/work/zephyr/zephyr/MAINTAINERS.yml: glob pattern 'modules/Kconfig.imx' in 'files' in area 'West project: hal_nxp' does not match any files
> The full traceback has been saved in /tmp/sphinx-err-zx0nospo.log, if you want to report the issue to the developers.
> Please also report this if it was a user error, so that a better error message can be provided next time.
> A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
> ```

I have updated this PR to fix this issue. PR (https://github.com/zephyrproject-rtos/zephyr/pull/84423) is a version that does not use the MCUX SDK NEXT kconfig tree, while this PR is a version that uses the MCUX SDK-NEXT kconfig tree. I have updated the titles of these two PRs to distinguish them.

hakehuang commented:
> Hello @hakehuang, I have fixed all twister issues reported by CI, could you please help test again? Thanks.

sample.display.shield	v4.0.0-4307-g23e82e911eea	Build failure	frdm_mcxn236/mcxn236
```
/home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/mipi_dbi/mipi_dbi_nxp_flexio_lcdif.c:434:29: error: initialization of 'void (*)(_Bool)' from incompatible pointer type 'void (*)(_Bool, void *)' [-Werror=incompatible-pointer-types] 434 | .setCSPin = flexio_lcdif_set_cs, \ | ^~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/sys/util_internal.h:72:26: note: in definition of macro '__DEBRACKET' 72 | #define __DEBRACKET(...) __VA_ARGS__ | ^~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET' 64 | __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code) | ^~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/sys/util_internal.h:59:9: note: in expansion of macro '__COND_CODE' 59 | __COND_CODE(_XXXX##_flag, _if_1_code, _else_code) | ^~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/sys/util_macro.h:204:9: note: in expansion of macro 'Z_COND_CODE_1' 204 | Z_COND_CODE_1(_flag, _if_1_code, _else_code) | ^~~~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/devicetree.h:4956:9: note: in expansion of macro 'COND_CODE_1' 4956 | COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT), \ | ^~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/twister-out/frdm_mcxn236_mcxn236/zephyr/samples/drivers/display/sample.display.shield/zephyr/include/generated/zephyr/devicetree_generated.h:20829:60: note: in expansion of macro 'MCUX_FLEXIO_LCDIF_DEVICE_INIT' 20829 | #define DT_FOREACH_OKAY_INST_nxp_mipi_dbi_flexio_lcdif(fn) fn(0) | ^~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_FOREACH_OKAY_INST_nxp_mipi_dbi_flexio_lcdif' 105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__ | ^ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/mipi_dbi/mipi_dbi_nxp_flexio_lcdif.c:485:1: note: in expansion of macro 'DT_INST_FOREACH_STATUS_OKAY' 485 | DT_INST_FOREACH_STATUS_OKAY(MCUX_FLEXIO_LCDIF_DEVICE_INIT) | ^~~~~~~~~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/mipi_dbi/mipi_dbi_nxp_flexio_lcdif.c:434:29: note: (near initialization for 'flexio_mculcd_0.setCSPin') 434 | .setCSPin = flexio_lcdif_set_cs, \ | ^~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/sys/util_internal.h:72:26: note: in definition of macro '__DEBRACKET' 72 | #define __DEBRACKET(...) __VA_ARGS__ | ^~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET' 64 | __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code) | ^~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/sys/util_internal.h:59:9: note: in expansion of macro '__COND_CODE' 59 | __COND_CODE(_XXXX##_flag, _if_1_code, _else_code) | ^~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/sys/util_macro.h:204:9: note: in expansion of macro 'Z_COND_CODE_1' 204 | Z_COND_CODE_1(_flag, _if_1_code, _else_code) | ^~~~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/devicetree.h:4956:9: note: in expansion of macro 'COND_CODE_1' 4956 | COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT), \ | ^~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/twister-out/frdm_mcxn236_mcxn236/zephyr/samples/drivers/display/sample.display.shield/zephyr/include/generated/zephyr/devicetree_generated.h:20829:60: note: in expansion of macro 'MCUX_FLEXIO_LCDIF_DEVICE_INIT' 20829 | #define DT_FOREACH_OKAY_INST_nxp_mipi_dbi_flexio_lcdif(fn) fn(0) | ^~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_FOREACH_OKAY_INST_nxp_mipi_dbi_flexio_lcdif' 105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__ | ^ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/mipi_dbi/mipi_dbi_nxp_flexio_lcdif.c:485:1: note: in expansion of macro 'DT_INST_FOREACH_STATUS_OKAY' 485 | DT_INST_FOREACH_STATUS_OKAY(MCUX_FLEXIO_LCDIF_DEVICE_INIT) | ^~~~~~~~~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/mipi_dbi/mipi_dbi_nxp_flexio_lcdif.c:435:29: error: initialization of 'void (*)(_Bool)' from incompatible pointer type 'void (*)(_Bool, void *)' [-Werror=incompatible-pointer-types] 435 | .setRSPin = flexio_lcdif_set_rs, \ | ^~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/sys/util_internal.h:72:26: note: in definition of macro '__DEBRACKET' 72 | #define __DEBRACKET(...) __VA_ARGS__ | ^~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET' 64 | __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code) | ^~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/sys/util_internal.h:59:9: note: in expansion of macro '__COND_CODE' 59 | __COND_CODE(_XXXX##_flag, _if_1_code, _else_code) | ^~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/sys/util_macro.h:204:9: note: in expansion of macro 'Z_COND_CODE_1' 204 | Z_COND_CODE_1(_flag, _if_1_code, _else_code) | ^~~~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/devicetree.h:4956:9: note: in expansion of macro 'COND_CODE_1' 4956 | COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT), \ | ^~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/twister-out/frdm_mcxn236_mcxn236/zephyr/samples/drivers/display/sample.display.shield/zephyr/include/generated/zephyr/devicetree_generated.h:20829:60: note: in expansion of macro 'MCUX_FLEXIO_LCDIF_DEVICE_INIT' 20829 | #define DT_FOREACH_OKAY_INST_nxp_mipi_dbi_flexio_lcdif(fn) fn(0) | ^~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_FOREACH_OKAY_INST_nxp_mipi_dbi_flexio_lcdif' 105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__ | ^ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/mipi_dbi/mipi_dbi_nxp_flexio_lcdif.c:485:1: note: in expansion of macro 'DT_INST_FOREACH_STATUS_OKAY' 485 | DT_INST_FOREACH_STATUS_OKAY(MCUX_FLEXIO_LCDIF_DEVICE_INIT) | ^~~~~~~~~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/mipi_dbi/mipi_dbi_nxp_flexio_lcdif.c:435:29: note: (near initialization for 'flexio_mculcd_0.setRSPin') 435 | .setRSPin = flexio_lcdif_set_rs, \ | ^~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/sys/util_internal.h:72:26: note: in definition of macro '__DEBRACKET' 72 | #define __DEBRACKET(...) __VA_ARGS__ | ^~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET' 64 | __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code) | ^~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/sys/util_internal.h:59:9: note: in expansion of macro '__COND_CODE' 59 | __COND_CODE(_XXXX##_flag, _if_1_code, _else_code) | ^~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/sys/util_macro.h:204:9: note: in expansion of macro 'Z_COND_CODE_1' 204 | Z_COND_CODE_1(_flag, _if_1_code, _else_code) | ^~~~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/devicetree.h:4956:9: note: in expansion of macro 'COND_CODE_1' 4956 | COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT), \ | ^~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/twister-out/frdm_mcxn236_mcxn236/zephyr/samples/drivers/display/sample.display.shield/zephyr/include/generated/zephyr/devicetree_generated.h:20829:60: note: in expansion of macro 'MCUX_FLEXIO_LCDIF_DEVICE_INIT' 20829 | #define DT_FOREACH_OKAY_INST_nxp_mipi_dbi_flexio_lcdif(fn) fn(0) | ^~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_FOREACH_OKAY_INST_nxp_mipi_dbi_flexio_lcdif' 105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__ | ^ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/mipi_dbi/mipi_dbi_nxp_flexio_lcdif.c:485:1: note: in expansion of macro 'DT_INST_FOREACH_STATUS_OKAY' 485 | DT_INST_FOREACH_STATUS_OKAY(MCUX_FLEXIO_LCDIF_DEVICE_INIT) | ^~~~~~~~~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/mipi_dbi/mipi_dbi_nxp_flexio_lcdif.c:436:31: error: initialization of 'void (*)(_Bool)' from incompatible pointer type 'void (*)(_Bool, void *)' [-Werror=incompatible-pointer-types] 436 | .setRDWRPin = flexio_lcdif_set_rd_wr, \ | ^~~~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/sys/util_internal.h:72:26: note: in definition of macro '__DEBRACKET' 72 | #define __DEBRACKET(...) __VA_ARGS__ | ^~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET' 64 | __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code) | ^~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/sys/util_internal.h:59:9: note: in expansion of macro '__COND_CODE' 59 | __COND_CODE(_XXXX##_flag, _if_1_code, _else_code) | ^~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/sys/util_macro.h:204:9: note: in expansion of macro 'Z_COND_CODE_1' 204 | Z_COND_CODE_1(_flag, _if_1_code, _else_code) | ^~~~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/devicetree.h:4956:9: note: in expansion of macro 'COND_CODE_1' 4956 | COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT), \ | ^~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/twister-out/frdm_mcxn236_mcxn236/zephyr/samples/drivers/display/sample.display.shield/zephyr/include/generated/zephyr/devicetree_generated.h:20829:60: note: in expansion of macro 'MCUX_FLEXIO_LCDIF_DEVICE_INIT' 20829 | #define DT_FOREACH_OKAY_INST_nxp_mipi_dbi_flexio_lcdif(fn) fn(0) | ^~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_FOREACH_OKAY_INST_nxp_mipi_dbi_flexio_lcdif' 105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__ | ^ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/mipi_dbi/mipi_dbi_nxp_flexio_lcdif.c:485:1: note: in expansion of macro 'DT_INST_FOREACH_STATUS_OKAY' 485 | DT_INST_FOREACH_STATUS_OKAY(MCUX_FLEXIO_LCDIF_DEVICE_INIT) | ^~~~~~~~~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/mipi_dbi/mipi_dbi_nxp_flexio_lcdif.c:436:31: note: (near initialization for 'flexio_mculcd_0.setRDWRPin') 436 | .setRDWRPin = flexio_lcdif_set_rd_wr, \ | ^~~~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/sys/util_internal.h:72:26: note: in definition of macro '__DEBRACKET' 72 | #define __DEBRACKET(...) __VA_ARGS__ | ^~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET' 64 | __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code) | ^~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/sys/util_internal.h:59:9: note: in expansion of macro '__COND_CODE' 59 | __COND_CODE(_XXXX##_flag, _if_1_code, _else_code) | ^~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/sys/util_macro.h:204:9: note: in expansion of macro 'Z_COND_CODE_1' 204 | Z_COND_CODE_1(_flag, _if_1_code, _else_code) | ^~~~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/devicetree.h:4956:9: note: in expansion of macro 'COND_CODE_1' 4956 | COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT), \ | ^~~~~~~~~~~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/twister-out/frdm_mcxn236_mcxn236/zephyr/samples/drivers/display/sample.display.shield/zephyr/include/generated/zephyr/devicetree_generated.h:20829:60: note: in expansion of macro 'MCUX_FLEXIO_LCDIF_DEVICE_INIT' 20829 | #define DT_FOREACH_OKAY_INST_nxp_mipi_dbi_flexio_lcdif(fn) fn(0) | ^~ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_FOREACH_OKAY_INST_nxp_mipi_dbi_flexio_lcdif' 105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__ | ^ /home/jenkins/agent/workspace/bitbucket_build_platform/zephyr/drivers/mipi_dbi/mipi_dbi_nxp_flexio_lcdif.c:485:1: note: in expansion of macro 'DT_INST_FOREACH_STATUS_OKAY' 485 | DT_INST_FOREACH_STATUS_OKAY(MCUX_FLEXIO_LCDIF_DEVICE_INIT) | ^~~~~~~~~~~~~~~~~~~~~~~~~~~ cc1: all warnings being treated as errors ninja: build stopped: subcommand failed. 
```
